server/auctionapp/src/main/java/com/atlantbh/auctionapp/service/ProductService.java

```diff
@@ -74,6 +75,12 @@ public PriceRangeProj getPriceRange(){
     public ProductEntity createProduct(ProductRequest productRequest) {
         CategoryEntity category = categoryRepository.findById(productRequest.getCategoryId()).orElseThrow(() -> new NotFoundException("Category with id: " + productRequest.getCategoryId() + " does not exist"));
         ProductEntity product = new ProductEntity(productRequest.getProductName(), productRequest.getDescription(), productRequest.getStartPrice(), productRequest.getStartDate(), productRequest.getEndDate(), productRequest.getImages(), productRequest.getAddress(), productRequest.getCity(), productRequest.getZipCode(), productRequest.getCountry(), productRequest.getPhone(), productRequest.getUserId(), category);
+        if (productRequest.getEndDate().isBefore(LocalDateTime.now()))
+            throw new BadRequestException("End date can't be before current date");
+        if (!productRequest.getEndDate().isAfter(productRequest.getStartDate()))
```

Comment:
perhaps `productRequest.getEndDate().isBefore(productRequest.getStartDate())` is more intuitive, and more readable.